documentation/src/main/asciidoc/user_guide/query.adoc

```diff
@@ -92,73 +144,468 @@ for (Object book : objectList) {
+
+[source,xml]
+----
+    ...
+    <indexing index="ALL">
+        <property name="org.infinispan.sample.Car.directory_provider">ram</property>
+    </indexing>
+    ...
+</infinispan>
+
+----
+
+[source,java]
+----
+cacheCfg.indexing()
+   .index(Index.ALL)
+      .addProperty("org.infinispan.sample.Car.directory_provider", "ram")
+
+----
+
+Infinispan creates an index for each entity existent in a cache, and it allows to configure those indexes independently.
+For a class annotated with `@Indexed`, the default index name is the fully qualified class name, unless overridden with the
+`name` argument in the annotation.
+
+Example: The snippet below shows indexing configuration with the default index storage on `infinispan`, but `Boat` instances will be stored on `ram` in an index named
+`boatIndex`. `Airplane` entities will also be stored in `ram`. Any other entity's index will be configured with the property prefixed by `default`.
+
+[source,java]
+----
+package org.infinispan.sample;
+
+@Indexed(name = "boatIndex")
+public class Boat {
+
+}
+
+@Indexed
+public class Airplane {
+
+}
+----
+
+[source,xml]
+----
+    ...
+    <indexing index="ALL">
+        <property name="default.directory_provider">infinispan</property>
+        <property name="boatIndex.directory_provider">ram</property>
+        <property name="org.infinispan.sample.Airplane.directory_provider">
+            ram
+        </property>
+    </indexing>
+    ...
+</infinispan>
+
+----
+
+
+====== Specifying indexed Entities
+
+Infinispan can automatically recognize and manage indexes for different entity types in a cache. Future versions of Infinispan will remove this capability so it's recommended to
+declare upfront which types are going to be indexed. This can be done via xml:
 
 [source,xml]
 ----
 <infinispan>
    <cache-container default-cache="default">
-      <local-cache name="default">
-         <indexing index="LOCAL">
-            <property name="default.directory_provider">ram</property>
+      <replicated-cache name="default">
+         <indexing index="ALL">
+            <indexed-entities>
+                <indexed-entity>org.infinispan.query.test.Car</indexed-entity>
+                <indexed-entity>org.infinispan.query.test.Truck</indexed-entity>
+            </indexed-entities>
          </indexing>
-      </local-cache>
+      </replicated-cache>
    </cache-container>
 </infinispan>
 
 ----
 
-In this example the index is stored in memory, so when this nodes is shutdown the index is lost: good for a quick demo, but in real world cases you'll want to use the default (store on filesystem) or store the index in Infinispan as well. For the complete reference of properties to define, refer to the link:http://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#search-configuration[Hibernate Search documentation] .
+or programmatically:
+
+[source,java]
+----
+ cacheCfg.indexing()
+      .index(Index.ALL)
```

Comment:
I agree with you, in fact this config is called `IndexMode` internally. 

Since we're going final in 2 days, we'll keep as it is for now.  Going forward, there's a plan to deprecated this config and downgrade it to index manager level, since a cache can have more than one entity type, each with its own IndexManager with a different requirement of IndexMode.